run-many appends a "--" argument when invoking packages' scripts

## Current Behavior
In a monorepo where each package has a `tsc` script to run the Typescript checker, when I run `tsc` through Nx using this command:

```
nx run-many --target=tsc --all --skip-nx-cache
```

Nx appends `--` when it invokes the scripts, and that results in the following error:

```
> nx-bug@1.0.0 tsc /Users/pepicrft/Downloads/nx-bug
> nx run-many --target=tsc --all --skip-nx-cache



    ✖  nx run project:tsc
       > project@1.0.0 tsc /Users/pepicrft/Downloads/nx-bug/packages/project
       > tsc -p './tsconfig.json' "--"
       
       error TS5023: Unknown compiler option '--'.
 ```

## Expected Behavior
I'd expect Nx to invoke the script as it is declared in the `package` without appending additional arguments.

## Steps to Reproduce
1. Download this project [nx-bug.zip](https://github.com/nrwl/nx/files/8609378/nx-bug.zip)
2. Install dependencies `pnpm install`.
3. Run `pnpm run tsc`

### Failure Logs
The code is included above

### Environment
```
Node : 16.13.2
   OS   : darwin arm64
   pnpm : 7.0.0
   
   nx : 14.0.5
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.0.5
   @nrwl/eslint-plugin-nx : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.0.5
   @nrwl/js : Not Found
   @nrwl/linter : 14.0.5
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/workspace : 14.0.5
   typescript : 4.6.4
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
 ```

--- a/packages/nx/src/utils/package-manager.ts
+++ b/packages/nx/src/utils/package-manager.ts
@@ -6,7 +6,7 @@ import { dirSync } from 'tmp';
 import { promisify } from 'util';
 import { readJsonFile, writeJsonFile } from './fileutils';
 import { PackageJson } from './package-json';
-import { gte } from 'semver';
+import { gte, lt } from 'semver';
 
 const execAsync = promisify(exec);
 
@@ -63,10 +63,8 @@ export function getPackageManagerCommand(
     }),
     pnpm: () => {
       const pnpmVersion = getPackageManagerVersion('pnpm');
-      let useExec = false;
-      if (gte(pnpmVersion, '6.13.0')) {
-        useExec = true;
-      }
+      const useExec = gte(pnpmVersion, '6.13.0');
+      const includeDoubleDashBeforeArgs = lt(pnpmVersion, '7.0.0');
       return {
         install: 'pnpm install --no-frozen-lockfile', // explicitly disable in case of CI
         ciInstall: 'pnpm install --frozen-lockfile',
@@ -75,7 +73,10 @@ export function getPackageManagerCommand(
         addGlobal: 'pnpm add -g',
         rm: 'pnpm rm',
         exec: useExec ? 'pnpm exec' : 'pnpx',
-        run: (script: string, args: string) => `pnpm run ${script} -- ${args}`,
+        run: (script: string, args: string) =>
+          includeDoubleDashBeforeArgs
+            ? `pnpm run ${script} -- ${args}`
+            : `pnpm run ${script} ${args}`,
         list: 'pnpm ls --depth 100',
       };
     },

